src/components/Forum/ForumPost/RenderPostScreen.js

```diff
@@ -65,86 +71,169 @@ const RenderPostScreen = ({ renderScreenType }) => {
         );
   }
 
+  function handleAddImage() {
+    if (post.img.length === 10) {
+      Alert.alert("You Can Only Upload A Maximum Of 10 Images");
+      return;
+    }
+
+    pickImageFromLibrary((uri) =>
+      uploadImage(
+        uri,
+        () => {},
+        () => {},
+        (src) => setPost({ ...post, img: [...post.img, src] }),
+        "Image Added Successfully"
+      )
+    );
+  }
+
+  function handleDeleteImage() {
+    //Users shouldn't know about this background process
+    removeImageFromCloudStorage(post.img[imageIndex]);
```

Comment:
I think the image should be removed from firebase storage when the confirmation button is clicked. In handleDeleteImage, you can store the image-to-be-deleted in some array. All images at this array will then be deleted in handleSubmit.